app/scripts/controllers/transactions/index.js

```diff
@@ -2121,6 +2167,7 @@ export default class TransactionController extends EventEmitter {
       asset_type: assetType,
       token_standard: tokenStandard,
       transaction_type: transactionType,
+      transaction_approval_amount_type: transactionApprovalAmountType,
```

Comment:
Can you only add this property if `transactionContractMethod === 'Approve'`?

Reply:
@danjm Done [here](https://github.com/MetaMask/metamask-extension/pull/15289/commits/cbebce7386de9fa462921da06c1af843b7e38939).